### PR TITLE
Added forecast quantiles for Single backtest

### PR DESCRIPTION
**Pull request by SamGos93**
# Description

# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by vlbejan — APPROVED** [review]

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/assets/retrain_models.py [hunk 1]
```diff
@@ -143,6 +153,15 @@ def run_backtest(data_input_name: str, file_name: str, experiment: Experiment):
     x_pred["actual_level"] = y_test
     x_pred["backtest_iteration"] = f"iteration_{last_training_date}"
     date_safe = RE_INVALID_SYMBOLS.sub("_", last_training_date)
+
+    # Adding the forecast quantile logic
+    if forecast_quantiles:
```
Comment (nick863) [inline v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/assets/retrain_models.py:158]: Let us move this part to line 139 and do:
```
# By default we will have forecast quantiles of 0.5, which is our target.
if forecast_quantiles:
    if 0.5 not in forecast_quantiles:
        forecast_quantiles.append(0.5)
    fitted_model.quantiles = forecast_quantiles
 x_pred = fitted_model.forecast_quantiles(X_test)
 x_pred.rename({0.5, "predicted_level"}, axis=1, inplace=True)
```
In this case we will not need to drop columns (remove lines 141-152)

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/auto-ml-forecasting-backtest-single-model.ipynb [hunk 2]
```diff
@@ -365,6 +365,7 @@
     "    step_size=BACKTESTING_PERIOD,\n",
     "    step_number=NUMBER_OF_BACKTESTS,\n",
     "    model_uid=model_uid,\n",
+    "    forecast_quantiles=[0.10, 0.90],\n",
```
Comment (nick863) [inline v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/auto-ml-forecasting-backtest-single-model.ipynb:368]: Should we set confidence intervals to [0.025, 0.975] as in  [orange juice notebook](https://github.com/Azure/azureml-examples/blob/sagoswami/local_quantile/v1/python-sdk/tutorials/automl-with-azureml/forecasting-orange-juice-sales/forecasting_script.py)?

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/auto-ml-forecasting-backtest-single-model.ipynb [hunk 3]
```diff
@@ -590,6 +591,7 @@
     "    step_size=BACKTESTING_PERIOD,\n",
     "    step_number=NUMBER_OF_BACKTESTS,\n",
     "    model_name=model_name,\n",
+    "    forecast_quantiles=[0.10, 0.90],\n",
```
Comment (nick863) [inline v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/auto-ml-forecasting-backtest-single-model.ipynb:594]: Should we set confidence intervals to [0.025, 0.975] as in [orange juice notebook](https://github.com/Azure/azureml-examples/blob/sagoswami/local_quantile/v1/python-sdk/tutorials/automl-with-azureml/forecasting-orange-juice-sales/forecasting_script.py)?

**Review by nick863 — CHANGES_REQUESTED** [review]

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/auto-ml-forecasting-backtest-single-model.ipynb (hunk 3 above)
Comment (SamGos93) [reply v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/auto-ml-forecasting-backtest-single-model.ipynb]: Done.

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/pipeline_helper.py [hunk 4]
```diff
@@ -135,6 +137,10 @@ def get_backtest_pipeline(
     if model_uid is not None:
         prs_args.append("--model-uid")
         prs_args.append(model_uid)
+    if forecast_quantiles is not None:
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/pipeline_helper.py]: Updated.Thanks.

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/assets/retrain_models.py (hunk 1 above)
Comment (SamGos93) [reply v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/assets/retrain_models.py]: Done. Thanks

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/assets/retrain_models.py [hunk 5]
```diff
@@ -143,6 +153,15 @@ def run_backtest(data_input_name: str, file_name: str, experiment: Experiment):
...
+        fitted_model.quantiles = forecast_quantiles
+        forecast_quantile_df = fitted_model.forecast_quantiles(X_test)
+        merge_keys = [automl_settings[constants.TimeSeries.TIME_COLUMN_NAME]]
+        merge_keys.extend([automl_settings[constants.TimeSeries.GRAIN_COLUMN_NAMES]])
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/assets/retrain_models.py]: As suggested, we are getting the forecasts from .forecast_quantailes(0.5). Hence, there is no merge required.

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/auto-ml-forecasting-backtest-single-model.ipynb [hunk 6]
```diff
@@ -714,11 +716,11 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.8.5"
+   "version": "3.7.13"
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/auto-ml-forecasting-backtest-single-model.ipynb]: Done!

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/pipeline_helper.py [hunk 7]
```diff
@@ -135,6 +137,10 @@ def get_backtest_pipeline(
     if model_uid is not None:
         prs_args.append("--model-uid")
         prs_args.append(model_uid)
+    if forecast_quantiles:
+        prs_args.append("--forecast_quantiles")
+        for val in range(len(forecast_quantiles)):
+            prs_args.append(forecast_quantiles[val])
```
Comment (iamrk04) [inline v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/pipeline_helper.py:143]: Let's simplify this:

prs_args.extend(forecast_quantiles)

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/assets/retrain_models.py [hunk 8]
```diff
@@ -126,23 +136,18 @@ def run_backtest(data_input_name: str, file_name: str, experiment: Experiment):
...
-        inplace=True,
-    )
+    # By default we will have forecast quantiles of 0.5, which is our target
+    if forecast_quantiles:
+        if 0.5 not in forecast_quantiles:
+            forecast_quantiles.append(0.5)
+        fitted_model.quantiles = forecast_quantiles
+
+    x_pred = fitted_model.forecast_quantiles(X_test)
+    x_pred.rename({0.5, "predicted_level"}, axis=1, inplace=True)
```
Comment (iamrk04) [inline v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/assets/retrain_models.py:146]: This requires rework.

Can fail when forecast_quantiles are not supplied. Also, we need to make sure that 0.5 is always in forecast_quantiles if we want to only rely on ``forecast_quantiles`` method.

**Review by iamrk04 — CHANGES_REQUESTED** [review]
I have added two comments, please have a look.

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/assets/retrain_models.py (hunk 8 above)
Comment (vlbejan) [reply v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/assets/retrain_models.py]: forecast_quantiles by default should provide the 0.5. Hence, there is no need to specify it.

Comment (iamrk04) [reply v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/assets/retrain_models.py]: resolving this as per confirmation from @vlbejan.

@SamGos93, please check the failing workflow.

File: v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/pipeline_helper.py (hunk 7 above)
Comment (SamGos93) [reply v1/python-sdk/tutorials/automl-with-azureml/forecasting-backtest-single-model/pipeline_helper.py]: Done. Thanks

**Review by nick863 — APPROVED** [review]

**Review by iamrk04 — APPROVED** [review]

**Pull request merged**